.github/workflows/test.yaml

```diff
@@ -48,31 +48,56 @@ jobs:
           python -m pip install -e .[develop,dash]
 
       - name: Test with pytest
-        run: python -m pytest
+        run: coverage run -p -m pytest
 
-      - uses: actions/cache@v3
+      - name: "Upload coverage data"
+        uses: actions/upload-artifact@v3
         with:
-          path: coverage.xml
-          key: ${{ github.sha }}-coverage
-
+          name: covdata
+          path: .coverage.*
 
-  sonar:
-    name: SonarCloud Upload
-    runs-on: ubuntu-latest
+  coverage:
+    name: coverage
     needs: test
-    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: "Check out the repo"
+        uses: "actions/checkout@v2"
```

Comment:
v2 is deprecated
```suggestion
        uses: "actions/checkout@v3"
```

Reply:
Thanks, I ran into this earlier. It's tricky to keep track of action versions :-)